Menus/TitleScreen.cpp

```diff
@@ -0,0 +1,608 @@
+
+		int starLargeBitmapCount = 1;
+		std::vector<BITMAP *> starLargeBitmaps = ConvertDoublePointerToVectorOfPointers(ContentFile("Base.rte/GUIs/Title/Stars/StarLarge.png").GetAsAnimation(starLargeBitmapCount), starLargeBitmapCount);
+
+		int starHugeBitmapCount = 2;
+		std::vector<BITMAP *> starHugeBitmaps = ConvertDoublePointerToVectorOfPointers(ContentFile("Base.rte/GUIs/Title/Stars/StarHuge.png").GetAsAnimation(starHugeBitmapCount), starHugeBitmapCount);
+
+		int starCount = (g_FrameMan.GetResX() * m_Nebula.GetBitmap()->h) / 1000;
+		for (int i = 0; i < starCount; ++i) {
+			Star newStar;
+			if (RandomNum() < 0.95F) {
+				newStar.Size = Star::StarSize::StarSmall;
+				newStar.Bitmap = starSmallBitmaps.at(RandomNum(0, starSmallBitmapCount - 1));
+				newStar.Intensity = RandomNum(0, 92);
+			} else if (RandomNum() < 0.85F) {
+				newStar.Size = Star::StarSize::StarLarge;
+				newStar.Bitmap = starLargeBitmaps.at(RandomNum(0, starLargeBitmapCount - 1));
+				newStar.Intensity = RandomNum(111, 185);
+			} else {
+				newStar.Size = Star::StarSize::StarHuge;
+				newStar.Bitmap = starHugeBitmaps.at(RandomNum(0, starLargeBitmapCount - 1));
+				newStar.Intensity = RandomNum(166, 185);
+			}
+			newStar.Position = Vector(RandomNum(0.0F, static_cast<float>(g_FrameMan.GetResX())), RandomNum(-100.0F, static_cast<float>(m_Nebula.GetBitmap()->h)));
+
+			m_BackdropStars.emplace_back(newStar);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::CreateIntroSequenceSlides() {
+		std::string highRes = (g_FrameMan.GetResY() >= 680) ? "HD" : "";
+		for (int slideNum = 0; slideNum < m_IntroSlides.size(); ++slideNum) {
+			m_IntroSlides.at(slideNum) = ContentFile(("Base.rte/GUIs/Title/Intro/IntroSlide" + std::to_string(slideNum + 1) + highRes + ".png").c_str()).GetAsBitmap(COLORCONV_NONE, false);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::Destroy() const {
+		for (BITMAP *slide : m_IntroSlides) {
+			if (slide) { destroy_bitmap(slide); }
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::Update() {
+		if (m_SectionSwitch) { m_SectionTimer.Reset(); }
+		m_SectionElapsedTime = static_cast<float>(m_SectionTimer.GetElapsedRealTimeS());
+		m_SectionProgress = std::min((m_SectionDuration > 0) ? m_SectionElapsedTime / m_SectionDuration : 0, 0.9999F);
+
+		if (m_StationOrbitProgress >= 0.999F) { m_StationOrbitTimer.Reset(); }
+		m_StationOrbitProgress = std::clamp(static_cast<float>(m_StationOrbitTimer.GetElapsedRealTimeS()) / 60.0F, 0.0F, 0.9999F);
+		m_StationOrbitRotation = LERP(0, 1.0F, c_PI, -c_PI, m_StationOrbitProgress);
+
+		if (m_FinishedPlayingIntro) {
+			UpdateTitleTransitions();
```

Comment:
Very minor nitpick, but it might be nice for method call order to match method order. That is to say, either UpdateTitleTransitions could be above the intro stuff (though that doesn't make much sense in terms of what's actually happening), or you could reverse the if condition and call UpdateTitleTransitions below the intro stuff.